contracts/pass-sale/CruzoPassSale.sol

```diff
@@ -0,0 +1,120 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import "../tokens/Cruzo1155.sol";
+
+import "../utils/Cruzo1155Factory.sol";
+
+contract CruzoPassSale is Ownable {
+    uint256 public constant MAX_PER_ACCOUNT = 3;
+    uint256 public constant REWARDS = 20;
+    uint256 public constant ALLOCATION = 180;
+    uint256 public constant MAX_SUPPLY = ALLOCATION + REWARDS;
+
+    address public tokenAddress;
+    address public signerAddress;
+    string[MAX_SUPPLY] private uris;
+    uint256 public price;
+
+    uint256 public tokenId;
+
+    mapping(address => uint256) public allocation;
+
+    bool public saleActive;
+    bool public publicSale;
+
+    event Mint(address to, uint256 tokenId);
+
+    constructor(
+        address _factoryAddress,
+        address _signerAddress,
+        address _rewardsAddress,
+        string[MAX_SUPPLY] memory _uris,
+        uint256 _price
+    ) {
+        tokenAddress = Cruzo1155Factory(_factoryAddress).create(
+            "CRUZO Collectors NFT Pass - OFFICIAL",
```

Comment:
I'd also pass it as params to constructor.

Reply:
I don't think it will be changed